Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -1,95 +1,183 @@
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            int[] newArray = new int[array.Length - 1];
+            
+            for (int i = 0; i < array.Length - 1; i++)
+            {
+                newArray[i] = array[i];
+            }
+            
+            return newArray;
         }
 
-        /// <summary>
-        /// Remove first element in array.
-        /// </summary>
-        /// <returns>A new array with the first element removed. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveFirst(int[] array)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            int[] newArray = new int[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i + 1];
+            }
+
+            return newArray;
         }
 
-        /// <summary>
-        /// Removes array element at given index.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="index">Index at which array element should be removed.</param>
-        /// <returns>A new array with element removed at a given index. If an array is empty or null, returns input array.</returns>
         public static int[] RemoveAt(int[] array, int index)
         {
-            // ToDo: implement.
-            return array;
+            if (IsNullOrEmpty(array))
+            {
+                return array;
+            }
+
+            int[] newArray = new int[array.Length - 1];
+
+            for (int i = 0; i < index; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            for (int i = index; i < array.Length - 1; i++)
+            {
+                newArray[i] = array[i + 1];
+            }
+
+            return newArray;
         }
 
-        /// <summary>
-        /// Inserts a new array element at the start.
-        /// </summary>
-        /// <param name="array">Input array.</param>
-        /// <param name="number">Number to be added.</param>
-        /// <returns>A new array with element added at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertFirst(int[] array, int number)
         {
-            // ToDo: implement.
-            return array;
+            int[] newArray = new int[array.Length + 1];
+
+            if (IsNullOrEmpty(array))
+            {
+                newArray[0] = number;
+                return newArray;
+            }
+
+            newArray[0] = number;
+
+            for (int i = 1; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i - 1];
+            }
+
+            return newArray;
         }
```

Comment:
Consider implementing this method in terms of `InsertAt` - hint: maybe this method could be replaced completely with `InsertAt(0)`